### PR TITLE
Add CODEOWNERS

### PR DESCRIPTION
**Pull request by withinfocus**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [X] Other
```

## Objective

Adds a `CODEOWNERS` file that asks for DevOps review for workflows.

## Code changes

- **.github/CODEOWNERS:** Single entry for DevOps.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Please sort into logical groups with comment headers. Sort groups in order of specificity.
+# For example, default owners should always be the first group.
+# Sort lines alphabetically within these groups to avoid accidentally adding duplicates.
+#
+# https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
+
+
+# DevOps for Actions and other workflow changes.
+.github/workflows @bitwarden/dept-devops
```
Comment (Hinton) [inline .github/CODEOWNERS:9]: Only the release pipelines are really owned by devops.

Comment (withinfocus) [reply .github/CODEOWNERS:9]: We discussed this with DevOps and they'd like general oversight of workflows altogether. This is a good best practice as they possess a lot of knowledge about GitHub workflows. Additionally, we have seen a lot of benefits come from collaboration -- the test repo being a good example.

Comment (Hinton) [reply .github/CODEOWNERS:9]: That's fine and we typically do request review from devops when it makes sense. However I don't want to block PRs requiring minor changes to the workflows before we can ensure the process works. (Check contributing docs)

Comment (withinfocus) [reply .github/CODEOWNERS:9]: This is in place for our biggest repos today. DevOps is available to review changes.

Comment (joseph-flinn) [reply .github/CODEOWNERS:9]: > However I don't want to block PRs requiring minor changes to the workflows before we can ensure the process works.

Is this referring to the Code Owner review process? or the process that is being written and updated in the GitHub Workflow itself?

Comment (Hinton) [reply .github/CODEOWNERS:9]: @joseph-flinn The review process. The workflows in this repo aren't stable (other than the publish) and we tend to make minor adjustments frequently. I mainly don't want to go **ping** devops all the time but if that's our process I'll do it.

Comment (joseph-flinn) [reply .github/CODEOWNERS:9]: Got it, thanks!

Long term, I'd like to see DevOps set up more robust tooling around workflow changes (mostly linting) to enable a better development experience for all engineering teams while working in the pipelines while also enforcing secure industry practices. I'll work with the team to have quick turn around times for workflow changes until we have capacity to build out the tooling.

**Review by Hinton — APPROVED** [review]

**Pull request merged**